instrumentation/opentelemetry-instrumentation-asgi/tests/test_asgi_custom_headers.py

```diff
@@ -88,6 +158,11 @@ def setUp(self):
         self.app = otel_asgi.OpenTelemetryMiddleware(
             simple_asgi, tracer_provider=self.tracer_provider
         )
+        self.previous_propagator = get_global_textmap()
+        set_global_textmap(MockTextMapPropagator())
+
+    def tearDown(self):
+        set_global_textmap(self.previous_propagator)
```

Comment:
I couldn't follow what the tests were targeting. I would test that `ASGIGetter` instance doesn't throw an exception for scope dict, which is currently failing, and additionally assert that keys return the Headers keys.

Reply:
The idea was to ensure that the values returned by `get()` and `keys()` were the same and they were consistent across different tests, but it was a bit of a hack indeed with that custom propagator. I added simpler tests in `test_getter.py` instead that indeed fail before the fix and pass now, hope they make sense https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1435/commits/c219bb0fd9884f7a7720a1192f480cef42379812